### PR TITLE
Fix and modify the test cases with fix done in wso2-axis2 #166

### PR DESCRIPTION
**Pull request by Vathsan**
## Purpose
Fix and modify failing test cases with the fix done in [wso2-axis2/pull/166](https://github.com/wso2/wso2-axis2/pull/166)

## Approach
1. Test case [`ESBJAVA_4239_HTTP_SC_HandlingTests.java`](https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/mediators/callout/ESBJAVA_4239_HTTP_SC_HandlingTests.java) fails because AxisFault will not be thrown for Http response code 404 after the fix mentioned above. Because both success and non-success response codes will be sent back to the client instead of throwing AxisFault as of non-blocking mode (description of behaviour before and after the fix can be found in [wso2-axis2/pull/166](https://github.com/wso2/wso2-axis2/pull/166).
To make the test case pass we need to throw AxisFault for Http status code 404. To throw AxisFault for Http status codes after the fix they must be set in [`error.http.status.codes`](https://github.com/wso2/wso2-axis2/blob/master/modules/kernel/src/org/apache/axis2/transport/http/HTTPConstants.java) property.

2. Test case [`ESBJAVA4423CustomStatusDescriptionTest.java`](https://github.com/wso2/product-ei/blob/master/integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java) is written to check a response with a custom status description with the fix done for the issue [ESBJAVA-4423](https://wso2.org/jira/browse/ESBJAVA-4423). In the written test case `sendSimpleStockQuoteRequest` method uses `CommonsHTTPTransportSender`, instead of that, we should be using Passthrough Transport to test the fix.

## Related Issues
https://wso2.org/jira/browse/ESBJAVA-4239
https://wso2.org/jira/browse/ESBJAVA-4423

## Related PRs
[wso2-axis2/pull/166](https://github.com/wso2/wso2-axis2/pull/166)

## Note
PR should be merged after merging [wso2-axis2/pull/166](https://github.com/wso2/wso2-axis2/pull/166) in axis2 (which fixes [product-ei/issues/951](https://github.com/wso2/product-ei/issues/951)) and upgrading the axis2 version in product-ei after the axis2 build.

### CODE REVIEW

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java [hunk 1]
```diff
@@ -55,12 +54,12 @@ public void testCustomStatusDescription() {
...
-            axis2Client.sendSimpleStockQuoteRequest(getProxyServiceURLHttp("ESBJAVA4423HttpCustomProxyTest"), "", "IBM");
-        } catch (IOException e) {
-            Assert.assertTrue(e.getMessage().contains("Custom response"));
-        }
+        String endpoint = getProxyServiceURLHttp("ESBJAVA4423HttpCustomProxyTest");
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "text/xml");
+
+        HttpResponse httpResponse = HttpRequestUtil.doGet(endpoint, header);
+        Assert.assertTrue(httpResponse.getResponseMessage().contains("Custom response"));
```
Comment (chanikag) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java]: Let's add an assert message

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java [hunk 2]
```diff
@@ -1,8 +1,8 @@
 /*
  * Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
  *
- * WSO2 Inc. licenses this file to you under the Apache License, 
- * Version 2.0 (the "License"); you may not use this file except 
+ * WSO2 Inc. licenses this file to you under the Apache License,
```
Comment (chanikag) [inline integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java]: It is better to revert this commit as this is not related to your change

File: integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java (hunk 1 above)
Comment (Vathsan) [reply integration/mediation-tests/tests-patches/src/test/java/org/wso2/carbon/esb/passthru/transport/test/ESBJAVA4423CustomStatusDescriptionTest.java]: Done

**Comment by Nirothipan** [comment]
Closing since this fix is on hold https://github.com/wso2/wso2-axis2/pull/166
